security/pkg/credentialfetcher/plugin/gce.go

```diff
@@ -47,14 +63,63 @@ func CreateGCEPlugin(audience, jwtPath, identityProvider string) *GCEPlugin {
 		jwtPath:          jwtPath,
 		identityProvider: identityProvider,
 	}
+	go p.startTokenRotationJob()
 	return p
 }
 
+func (p *GCEPlugin) Stop() {
+	p.closing <- true
+}
+
+func (p *GCEPlugin) startTokenRotationJob() {
+	// Wake up once in a while and refresh GCE VM credential.
+	p.rotationTicker = time.NewTicker(rotationInterval)
+	for {
+		select {
+		case <-p.rotationTicker.C:
+			p.rotate()
+		case <-p.closing:
+			if p.rotationTicker != nil {
+				p.rotationTicker.Stop()
+			}
+		}
+	}
+}
+
+func (p *GCEPlugin) rotate() {
+	if p.shouldRotate() {
+		_, err := p.GetPlatformCredential()
+		gcecredLog.Errorf("credential refresh failed: %+v", err)
+	}
+}
+
+func (p *GCEPlugin) shouldRotate() bool {
+	p.tokenMutex.RLock()
+	defer p.tokenMutex.RUnlock()
+
+	if p.tokenCache == "" {
+		return true
+	}
+	exp, err := util.GetExp(p.tokenCache)
+	// When fails to get expiration time from token, always refresh the token
+	if err != nil || exp.IsZero() {
+		return true
+	}
+	rotate := time.Now().After(exp.Add(-gracePeriod))
+	// TODO(JimmyCYJ): change this log to Debug level once token refresh is stable.
```

Comment:
why not just turn on debug logs? this will be pretty spammy

Reply:
Set it back to debugf now.